codec-http/src/test/java/io/netty/handler/codec/http/cors/CorsHandlerTest.java

```diff
@@ -289,17 +293,100 @@ public void shortCurcuitNonCorsRequest() {
     }
 
     @Test
+    public void shortCurcuitWithConnectionKeepAliveShouldStayOpen() {
+        final CorsConfig config = forOrigin("http://localhost:8080").shortCircuit().build();
+        final EmbeddedChannel channel = new EmbeddedChannel(new CorsHandler(config));
+        final FullHttpRequest request = createHttpRequest(GET);
+        request.headers().set(ORIGIN, "http://localhost:8888");
+        request.headers().set(CONNECTION, KEEP_ALIVE);
+
+        channel.writeInbound(request);
+        final HttpResponse response = channel.readOutbound();
+
+        assertThat(channel.isOpen(), is(true));
+        assertThat(response.status(), is(FORBIDDEN));
```

Comment:
also call `channel.finish()` and assert return value and ensure you release all the stuff in the `EmbeddedChannel`